emr-datahub/src/main/scala/org/apache/spark/sql/aliyun/datahub/DatahubContinuousReader.scala

```diff
@@ -170,8 +165,19 @@ class DatahubContinuousDataReader(
     rowWriter.write(1, UTF8String.fromString(topic))
     rowWriter.write(2, UTF8String.fromString(shardId))
     rowWriter.write(3, DateTimeUtils.fromJavaTimestamp(
-      new java.sql.Timestamp(currentRecord.get(DatahubSchema.SYSTEM_TIME).asInstanceOf[Long])))
-    rowWriter.write(4, currentRecord.toJSONString.getBytes)
+      new java.sql.Timestamp(currentRecord.getSystemTime)))
+
+    fields.getOrElse({
+      fields = Some(datahubClient.getTopic(project, topic).getRecordSchema.getFields.toList)
+      fields.get
```

Comment:
do a schema request to server for each record?

Reply:
no, the fields in schema will request from server and be stored when handling first record, and will reuse for the rest of records.